src/main/java/ome/formats/importer/transfers/SymlinkFileTransfer.java

```diff
@@ -63,9 +65,14 @@ protected ProcessBuilder createProcessBuilder(File file, File location) {
     }
 
     /**
-     * No cleanup is needed for soft-linking.
+     * Set permissions on original file to read-only.
      */
+    @Override
     public void afterTransfer(int errors, List<String> srcFiles) {
-        // no-op
+        if (errors == 0) {
+            for (final String srcFile : srcFiles) {
+                FileSystemUtil.setReadOnly(srcFile);
```

Comment:
Am I right in thinking that this is the only example of our client-side code mutating the state of imported files?

Reply:
Quite probably. One reason it's in a separate commit is in case we don't want it. :smiley: